### PR TITLE
Resolve issue: https://github.com/VTUL/fcrepo4-ansible/issues/31

### PR DESCRIPTION
**Pull request by yinlinchen**
# What does this Pull Request do? (:star:)
Update Ubuntu to 16.04 in AWS
Update Fedora to 4.7.5
Update syntax to remove deprecation warning

# How should this be tested?
Local test: vagrant up
AWS test: vagrant up --provider aws

# Interested parties
Tag (@pmather ) interested parties

### CODE REVIEW

File: Vagrantfile [hunk 1]
```diff
@@ -133,8 +133,8 @@ Vagrant.configure(VAGRANTFILE_API_VERSION) do |config|
       aws.access_key_id = ENV['AWS_ACCESS_KEY']
       aws.secret_access_key = ENV['AWS_SECRET_KEY']
       aws.keypair_name = keypair
-      aws.ami = "ami-e7ab8a9c" # Ubuntu Trusty LTS
-      aws.block_device_mapping = [{ 'DeviceName' => '/dev/sda1', 'Ebs.VolumeSize' => 500 }]
+      aws.ami = "ami-0f9cf087c1f27d9b1" # Ubuntu Server 16.04 LTS
```
Comment (pmather) [inline Vagrantfile]: You indicate you are using an Ubuntu 16.04 (Xenial) AMI for AWS but you still use Ubuntu 14.04 (Trusty) for the other providers (VirtualBox and OpenStack on Chameleon Cloud).  You should use Xenial throughout, i.e., use `override.vm.box = "ubuntu/trusty64"`  in the `:virtualbox` provider and use `os.image  = "CC-Ubuntu16.04" (I believe that's the correct image name) in the `: openstack` provider.

File: ansible/fcrepo4.yml [hunk 2]
```diff
@@ -1,6 +1,16 @@
 ---
+- name: dependency provisioning
+  hosts: all
+  become: yes
+  become_method: sudo
+  gather_facts: false
+  tasks:
+    - name: install python2
```
Comment (pmather) [inline ansible/fcrepo4.yml]: If you are installing Python 2 just to get Ansible to work to provision on a Xenial host then that is not necessary.  The [recommended way](https://docs.ansible.com/ansible/latest/reference_appendices/python_3_support.html#using-python-3-on-the-managed-machines-with-commands-and-playbooks) is to set the `ansible_python_interpreter` Ansible variable to `/usr/bin/python3`.

In Vagrant terms, this means you can add the following to your `config.vm.provision :ansible` section:
```
ansible.host_vars = {
   :fcrepo4 => {'ansible_python_interpreter' => '/usr/bin/python3'}
}
```

**Review by pmather — CHANGES_REQUESTED** [review]
I recommend you use Ubuntu Xenial 16.04 as the OS throughout and avoid installing Python 2 unnecessarily, as indicated in the separate comments.

File: ansible/fcrepo4.yml (hunk 2 above)
Comment (pmather) [reply ansible/fcrepo4.yml]: I am curious to know: what "other errors" would the above change introduce?

All currently supported versions of Ansible (2.5 onwards) support Python 3, and the above method is the recommended way of telling Ansible to use Python 3 on hosts that have only Python 3 installed.

If there is other Fedora-related software that requires Python 2 to be installed for Fedora to run then by all means install Python 2.  But, if you believe that Python 2 must be installed on Ubuntu Xenial for Ansible playbooks to run there then you are mistaken.  (The `ansible_python_interpreter` technique is used with `VTUL/InstallScripts`'s `Vagrantfile` when Xenial is the base OS.)

Installing unnecessary software is unnecessary.

Comment (yinlinchen) [reply ansible/fcrepo4.yml]: We still use python2 in our current local development, to have only python3 installed is not applied to us. It is out of the scope of this PR. Based on your input, this PR is updated to install Python 3 and added a note to indicate this in the readme.

**Review by pmather — APPROVED** [review]
This contains all the suggested changes.  I was able to deploy successfully a standard file-based database Fedora 4.7.5 both locally (via VirtualBox) and to AWS.  The resultant Fedora 4.7.5 appeared to work.  I was not able to test the PostgreSQL-backed Fedora 4.7.5.

**Pull request merged**